### ISSUE TITLE
I set rb manul,but the hardNodeAffinity not effective

### ISSUE THREAD

**Issue by 13567436138** [karmada-io/karmada#3189]
**What happened**:
```
apiVersion: work.karmada.io/v1alpha2
kind: ResourceBinding
metadata:
  name: nginx-deployment
spec:
  replicaRequirements:
    nodeClaim:
      hardNodeAffinity:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: NotIn
            values:
            - 192.168.229.154
            - 192.168.229.155
            - 192.168.229.157
            - 192.168.229.158
  clusters:
  - name: cluster1
    replicas: 2
  - name: cluster2
    replicas: 3
  replicas: 5
  resource:
    apiVersion: apps/v1
    kind: Deployment
    name: nginx
    namespace: karmada


apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - image: nginx
        name: nginx

```

```
[root@member1-153 ~]# kubectl get pod -n karmada -owide
NAME                                         READY   STATUS    RESTARTS        AGE     IP            NODE              NOMINATED NODE   READINESS GATES
nginx-85b98978db-9x484                       1/1     Running   0               6s      172.21.2.88   192.168.229.155   <none>           <none>
nginx-85b98978db-hdnhr                       1/1     Running   0               6s      172.21.0.65   192.168.229.154   <none>           <none>
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by chaunceyjiang** [comment]
https://github.com/karmada-io/karmada/blob/master/pkg/controllers/binding/binding_controller.go#L210-L227


Because you created the RB manually, it does not satisfy this rule.

**Comment by jwcesign** [comment]
Now `replicaRequirements` is used to selector clusters(by available resource), with your configuration, the available resource will not contain 192.168.229.154, 192.168.229.155,192.168.229.157,192.168.229.158. Haven't got a connection with the kind of NodeAffinity that you want.

If you want, just set NodeAffinity in the deployment yaml.
